src/main/java/com/example/nlern/controllers/subloadController.java

```diff
@@ -1,23 +1,132 @@
 package com.example.nlern.controllers;
 
-import com.example.nlern.models.User;
+import com.example.nlern.models.FullTest;
+import com.example.nlern.models.Question;
+import com.example.nlern.models.Test;
+
 import com.example.nlern.repos.GroupRepository;
+import com.example.nlern.repos.QuestionRepository;
+import com.example.nlern.repos.TestRepository;
+import com.example.nlern.repos.UserRepository;
+
 import org.springframework.beans.factory.annotation.Autowired;
-import org.springframework.web.bind.annotation.*;
+import org.springframework.http.ResponseEntity;
+import org.springframework.ui.Model;
+
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RestController;
+import org.springframework.web.bind.annotation.CrossOrigin;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.DeleteMapping;
 
-import java.util.HashMap;
-import java.util.Map;
+import java.util.List;
 
 @RestController
+@CrossOrigin
 @RequestMapping("/subload")
-public class subloadController {
+public class SubloadController {
+
+    @Autowired
+    private QuestionRepository questionRepository;
+
+    @Autowired
+    private TestRepository testRepository;
+
+    @Autowired
+    private GroupRepository groupRepository;
+
     @Autowired
-    private GroupRepository GroupRepo;
-    @PostMapping("/check_group")
-    public Map<String, Object> createUser(@RequestParam("group") String group){
+    private UserRepository userRepository;
+
+
+    @GetMapping("check_group/{name}")
+    public boolean checkGroupName(@PathVariable String name) {
+        return groupRepository.existsByName(name);
+    }
+
+    @GetMapping("teachersHome")
+    public List<Test> home(Model model) {
+        int teacherId = 3; //TODO get id from auth
+        return testRepository.findTop5ByAuthorIdOrderByIdDesc(teacherId);
+    }
+
+
+    @DeleteMapping("tests/{id}")
+    public ResponseEntity deleteClient(@PathVariable Integer id) {
+        questionRepository.deleteAllByTestId(id);
+        testRepository.deleteById(id);
+
+        return ResponseEntity.ok().build();
+    }
 
-        Map<String, Object> map = new HashMap<>();
-        map.put("groupExist", GroupRepo.existsByName(group));
-        return map;
+
+    @PostMapping("tests")
```

Comment:
Better prepend a slash everywhere so it's something like `/tests`

Reply:
Done